lua/neo-tree/sources/common/commands.lua

```diff
@@ -218,26 +218,42 @@ M.paste_from_clipboard = function(state, callback)
   end
 end
 
+---The root_directory is used to decide what part of the filename to show the
+-- user when asking for a new filename to e.g. copy or move to.
+local function get_using_root_directory(state, node)
+  local in_directory, _ = utils.split_path(node:get_id())
+  -- default to showing only the basename of the path
+  local using_root_directory = in_directory
+  if state.config.show_path == "absolute" then
+    using_root_directory = ""
+  elseif state.config.show_path == "relative" then
+    using_root_directory = state.path
+  end
+  return using_root_directory
+end
+
 ---Copies a node to a new location, using typed input.
 ---@param state table The state of the source
 ---@param callback function The callback to call when the command is done. Called with the parent node as the argument.
 M.copy = function(state, callback)
   local node = state.tree:get_node()
+  local using_root_directory = get_using_root_directory(state, node)
```

Comment:
Since you created the utility function, can you update `add` and `add_directory` to also use this `using_root_directory` function?